packages/core/src/amazonqGumby/telemetry/codeTransformTelemetry.ts

```diff
@@ -90,3 +105,79 @@ export const javapOutputToTelemetryValue = (javapCommandLineOutput: string) => {
 }
 
 export const calculateTotalLatency = (startTime: number): number => Date.now() - startTime
+
+type CodeTransformAPIresponse =
+    | CreateUploadUrlResponse
+    | ResumeTransformationResponse
+    | StopTransformationResponse
+    | GetTransformationPlanResponse
+    | StartTransformationResponse
+    | GetTransformationResponse
+
+export class TelemetryHelper {
+    private static get sessionId() {
+        return CodeTransformTelemetryState.instance.getSessionId()
+    }
+
+    static async callApi<T extends CodeTransformAPIresponse>(parameters: {
+        apiCall: () => Promise<PromiseResult<T, AWSError>>
+        apiName: CodeTransformApiNames
+        errorReason: string
+        uploadId?: string
+        jobId?: string
+        uploadFileByteSize?: number
+        setJobFailureMetadata: boolean
+        shouldAttachJobFailureMetadataOnError: boolean
+    }) {
+        const apiStartTime = Date.now()
+        let result = undefined
+        try {
+            result = await parameters.apiCall()
+            if (parameters.setJobFailureMetadata && result?.$response.requestId !== undefined) {
+                transformByQState.setJobFailureMetadata(` (request ID: ${result.$response.requestId})`)
+            }
+            return result
+        } catch (e: any) {
+            let errorMessage = (e as Error).message
+            getLogger().error(`CodeTransformation: ${parameters.apiName} error: = ${errorMessage}`)
+            if (parameters.shouldAttachJobFailureMetadataOnError) {
+                errorMessage += ` -- ${transformByQState.getJobFailureMetadata()}`
+            }
```

Comment:
Follow up with the above comment regarding "setJobFailureMetadata". Non-blocking, but looks like the failureMetadata is only used for appending the requestId to the errorMessage for telemetry (not even logging). However, the telemetry already have `codeTransformRequestId` field, so I don't think failureMetadata is needed and can be removed. For reference, IntelliJ doesn't append requestId to the end of error message.

Reply:
Yeah you are probably right, if we get this in I can remove this in a follow up pr tomorrow?